src/Symfony/Component/Scheduler/Trigger/PeriodicalTrigger.php

```diff
@@ -0,0 +1,150 @@
+            } else {
+                $a = (array) $interval;
+                $this->description = \PHP_VERSION_ID >= 80200 && $a['from_string'] ?? 'DateInterval';
+            }
+            if ($this->canBeConvertedToSeconds($i)) {
+                $this->intervalInSeconds = self::calcInterval($this->from, $this->from->add($i));
+            } else {
+                $this->period = new \DatePeriod($this->from, $i, $this->until);
+            }
+        } catch (\Exception $e) {
+            throw new InvalidArgumentException(sprintf('Invalid interval "%s": ', $interval instanceof \DateInterval ? 'instance of \DateInterval' : $interval).$e->getMessage(), 0, $e);
+        }
+    }
+
+    public function __toString(): string
+    {
+        return $this->description ?: sprintf('every %d seconds', $this->intervalInSeconds);
+    }
+
+    public function getNextRunDate(\DateTimeImmutable $run): ?\DateTimeImmutable
+    {
+        if ($this->intervalInSeconds) {
+            if ($this->from > $run) {
+                return $this->from;
+            }
+            if ($this->until <= $run) {
+                return null;
+            }
+
+            $delta = $run->format('U.u') - $this->from->format('U.u');
+            $recurrencesPassed = (int) ($delta / $this->intervalInSeconds);
+            $nextRunTimestamp = ($recurrencesPassed + 1) * $this->intervalInSeconds + $this->from->getTimestamp();
+            $nextRun = \DateTimeImmutable::createFromFormat('U.u', $nextRunTimestamp.$this->from->format('.u'), $this->from->getTimezone());
+
+            return $this->until > $nextRun ? $nextRun : null;
+        }
+
+        $iterator = $this->period->getIterator();
+        while ($run >= $next = $iterator->current()) {
+            $iterator->next();
+            if (!$iterator->valid()) {
+                return null;
+            }
+        }
+
+        return $next;
+    }
+
+    private function canBeConvertedToSeconds(\DateInterval $interval): bool
+    {
+        $a = (array) $interval;
+        if (\PHP_VERSION_ID >= 80200) {
+            if ($a['from_string']) {
+                return preg_match('#^\s*\d+\s*(sec|second|min|minute|hour)s?\s*$#', $a['date_string']);
+            }
+        } else {
+            if ($a['weekday'] || $a['weekday_behavior'] || $a['first_last_day_of'] || $a['days'] || $a['special_type'] || $a['special_amount'] || $a['have_weekday_relative'] || $a['have_special_relative']) {
+                return false;
+            }
+        }
+
+        return !$interval->y && !$interval->m && !$interval->d;
+    }
+
+    private static function calcInterval(\DateTimeImmutable $from, \DateTimeImmutable $to): int
+    {
+        if (8 <= \PHP_INT_SIZE) {
+            return $to->getTimestamp() - $from->getTimestamp();
+        }
+
+        // @codeCoverageIgnoreStart
+        $interval = $to->format('U') - $from->format('U');
+        self::ensureIntervalSize(abs($interval));
+
+        return (int) $interval;
```

Comment:
could this be a float somehow? on 32b archs, floats have more integer precision than int